Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -0,0 +1,1203 @@
+            }
+
+            // Do we have a source?
+            bool haveSource = !sourceAddr.Address.Equals(IPAddress.None);
+
+            if (haveSource)
+            {
+                if (!IsIP6Enabled && sourceAddr.AddressFamily == AddressFamily.InterNetworkV6)
+                {
+                    _logger.LogWarning("IPv6 is disabled in JellyFin, but enabled in the OS. This may affect how the interface is selected.");
+                }
+
+                if (!IsIP4Enabled && sourceAddr.AddressFamily == AddressFamily.InterNetwork)
+                {
+                    _logger.LogWarning("IPv4 is disabled in JellyFin, but enabled in the OS. This may affect how the interface is selected.");
+                }
+            }
+
+            bool isExternal = haveSource && !IsInLocalNetwork(sourceAddr);
+
+            string bindPreference = string.Empty;
+            if (haveSource)
+            {
+                // Check for user override.
+                foreach (var addr in _overrideUrls)
+                {
+                    // Remaining. Match anything.
+                    if (addr.Key.Equals(IPAddress.Broadcast))
+                    {
+                        bindPreference = addr.Value;
+                        break;
+                    }
+                    else if ((addr.Key.Equals(IPAddress.Any) || addr.Key.Equals(IPAddress.IPv6Any)) && (isExternal || chromeCast))
+                    {
+                        // External.
+                        bindPreference = addr.Value;
+                        break;
+                    }
+                    else if (addr.Key.Contains(sourceAddr))
+                    {
+                        // Match ip address.
+                        bindPreference = addr.Value;
+                        break;
+                    }
+                }
+            }
+
+            _logger.LogDebug("GetBindInterface: Souce: {0}, External: {1}:", haveSource, isExternal);
+
+            if (!string.IsNullOrEmpty(bindPreference))
+            {
+                // Has it got a port defined?
+                var parts = bindPreference.Split(':');
+                if (parts.Length > 1)
+                {
+                    if (int.TryParse(parts[1], out int p))
+                    {
+                        bindPreference = parts[0];
+                        port = p;
+                    }
+                }
+
+                _logger.LogInformation("{0}: Using BindAddress {1}:{2}", sourceAddr, bindPreference, port);
+                return bindPreference;
+            }
+
+            string ipresult;
+
+            // No preference given, so move on to bind addresses.
+            lock (_intLock)
+            {
+                var nc = _bindAddresses.Exclude(_bindExclusions).Where(p => !p.IsLoopback());
+
+                int count = nc.Count();
+                if (count == 1 && (_bindAddresses[0].Equals(IPAddress.Any) || _bindAddresses.Equals(IPAddress.IPv6Any)))
```

Comment:
```suggestion
                if (count == 1 && (_bindAddresses[0].Equals(IPAddress.Any) || _bindAddresses.Contains(IPAddress.IPv6Any)))
```

Reply:
Good catch - but it should be [0].Equals